lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -2329,6 +2329,118 @@ LogicalResult ConvertAtenOp<AtenThresholdOp>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenUnsqueezeOp>::matchAndRewrite(
+    AtenUnsqueezeOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  // Not a tensor type.
+  auto selfType = adaptor.self().getType().dyn_cast<TensorType>();
+  if (!selfType) {
+    return op.emitError("Only tensor types are currently supported");
+  }
+
+  auto selfRank = selfType.getRank();
+  auto selfElemTy = selfType.getElementType();
+  if (!selfElemTy.isIntOrFloat()) {
+    return op.emitError(
+        "Only floating-point or integer datatype legalization supported");
+  }
+
+  int64_t dim;
+  if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+    return op->emitError("dim must be a Scalar constant");
+
+  dim = toPositiveDim(dim, selfRank);
+  if (!isValidDim(dim, selfRank))
+    return op.emitError("dim is statically invalid");
+
+  SmallVector<int64_t> outShape;
+  for (auto en : llvm::enumerate(selfType.getShape())) {
+    if (static_cast<int64_t>(en.index()) == dim)
+      outShape.push_back(1);
+
+    outShape.push_back(en.value());
+  }
+
+  rewriter.replaceOpWithNewOp<tosa::ReshapeOp>(
+      op, getTypeConverter()->convertType(op.getType()), adaptor.self(),
+      rewriter.getI64ArrayAttr(outShape));
+
+  return success();
+}
+
+template <>
+LogicalResult ConvertAtenOp<AtenContiguousOp>::matchAndRewrite(
+    AtenContiguousOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  // Not a tensor type.
+  auto selfType = adaptor.self().getType().dyn_cast<TensorType>();
+  if (!selfType)
+    return op.emitError("Only tensor types are currently supported");
+
+  // FIXME: memory_format is not handled.
+
+  rewriter.replaceOp(op, adaptor.self());
+
+  return success();
+}
+
+template <>
+LogicalResult ConvertAtenOp<AtenDropoutOp>::matchAndRewrite(
+    AtenDropoutOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  // Not a tensor type.
+  auto selfType = adaptor.input().getType().dyn_cast<TensorType>();
+  if (!selfType)
+    return op.emitError("Only tensor types are currently supported");
+
+  // FIXME: train and p are not handled.
+
+  if (!op.train().getType().isa<Torch::NoneType>()) {
```

Comment:
the `if` is not needed because `train` can only be bool type.

Reply:
Incorporated in the latest push.